ddl/db_partition_test.go

```diff
@@ -3156,3 +3156,16 @@ func (s *testIntegrationSuite7) TestAddPartitionForTableWithWrongType(c *C) {
 	c.Assert(err, NotNil)
 	c.Assert(ddl.ErrWrongTypeColumnValue.Equal(err), IsTrue)
 }
+
+func (s *testIntegrationSuite5) TestBuildPartitionWithExpr(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
```

Comment:
there is a test `TestCreateTableWithHashPartition`, we can put the test into it.  BTW, if we just test expression in partition, we can also use the `list` or `range` partition to test it since they are `table drive` test.